app.js

```diff
@@ -2,108 +2,104 @@
+            label:"person",
+            id:"thomas",
+            firstName:"Thomas",
+            age:44, userid: 1
+        }).then(function (result) {
+                console.log("Result: %s\n", JSON.stringify(result));
+        });
+    }
 
-function addVertex2(callback)
+function addVertex2()
 {
-    console.log('Running Add Vertex2'); 
-    client.execute("g.addV(label).property('id', id).property('firstName', firstName).property('lastName', lastName).property('age', age).property('userid', userid)", 
-    { label:"person", id:"mary", firstName:"Mary", lastName: "Andersen", age:39, userid: 2}, 
-    (err, results) => {
-        if (err) {
-            return callback(console.error(err));
-        }
-
-        console.log("Result: %s\n", JSON.stringify(results));
-        callback(null)
-    });
+    console.log('Running Add Vertex2');
+    return client.submit("g.addV(label).property('id', id).property('firstName', firstName).property('lastName', lastName).property('age', age).property('userid', userid)", { 
+            label:"person",
+            id:"mary",
+            firstName:"Mary",
+            lastName: "Andersen",
+            age:39,
+            userid: 2
+        }).then(function (result) {
+            console.log("Result: %s\n", JSON.stringify(result));
+        });
 }
 
-function addEdge(callback)
+function addEdge()
 {
     console.log('Running Add Edge'); 
-    client.execute("g.V(source).addE(relationship).to(g.V(target))", 
-    {source:"thomas", relationship:"knows", target:"mary"}, 
-    (err, results) => {
-        if (err) {
-            return callback(console.error(err));
-        }
-
-        console.log("Result: %s\n", JSON.stringify(results));
-        callback(null)
-    });
+    return client.submit("g.V(source).addE(relationship).to(g.V(target))", {
+            source:"thomas", 
+            relationship:"knows", 
+            target:"mary"
+        }).then(function (result) {
+            console.log("Result: %s\n", JSON.stringify(result));
+        });
 }
 
-function countVertices(callback)
+function countVertices()
 {
     console.log('Running Count');
-    client.execute("g.V().count()", { }, (err, results) => {
-    if (err) {
-        return callback(console.error(err));
-    }
-
-    console.log("Result: %s\n", JSON.stringify(results));
-    callback(null)
-});
+    return client.submit("g.V().count()", { }).then(function (result) {
+        console.log("Result: %s\n", JSON.stringify(result));
+    });
 }
 
-function finish(err, result)
+function finish()
 {
-    if (err) {
-        return console.error(err);
-    }
     console.log("Finished");
     console.log('Press any key to exit');
     
-    process.stdin.setRawMode(true);
     process.stdin.resume();
     process.stdin.on('data', process.exit.bind(process, 0));
 }
 
-async.waterfall([
-    dropGraph,
-    addVertex1,
-    addVertex2,
-    addEdge,
-    countVertices
-], finish);
+client.open()
+    .then((res) => 
+        dropGraph()

```

Comment:
only some of these actually has a response when you're chaining them. I'd recommend just simplifying this and passing the method itself to then:

.then(dropGraph)
.then(addVertex1)

etc...

Reply:
Is this idiomatic? Since in the last iteration we used the `async.waterfall` method that looked a lot cleaner.